src/darwin/Framework/CHIP/MTRKeypair.h

```diff
@@ -31,13 +32,19 @@ NS_ASSUME_NONNULL_BEGIN
  * framework APIs.
  */
 @protocol MTRKeypair <NSObject>
-@required
+
+@optional
 /**
- * @brief Return public key for the keypair.
+ * @brief Returns a copy of the public key for the keypair.
  */
-- (SecKeyRef)publicKey;
+- (SecKeyRef)copyPublicKey MTR_NEWLY_AVAILABLE;
+
+/**
+ * @brief Returns public key for the keypair. DEPRECATED - please use copyPublicKey, otherwise this will leak
```

Comment:
```suggestion
 * @brief Returns public key for the keypair without adding a reference. DEPRECATED - please use copyPublicKey, otherwise this will leak.
```